pkg/proxy/proxy.go

```diff
@@ -185,32 +205,81 @@ func (p *Proxy) forwardRequest(host string, w http.ResponseWriter, req *http.Req
 		// If the client sent multiple XFF headers, flatten them.
 		proxyReq.Header.Set(xff, strings.Join(previous, ", "))
 	}
-	proxyReq.URL.Host = host
 	proxyReq.URL.Scheme = "https"
+	attempts := 0
 
-	log.Debug("Forwarding request to %s", proxyReq.URL.String())
-	client := http.Client{}
-	resp, err := client.Do(proxyReq)
-	if err != nil {
-		if urlErr, ok := err.(*url.Error); ok && urlErr.Timeout() {
-			writeJSONError(w, http.StatusGatewayTimeout, urlErr)
-		} else {
+	for {
+		proxyReq.URL.Host = acct.Host
+		log.Debug("Forwarding request to %s", proxyReq.URL.String())
+		client := http.Client{}
+		result, err := client.Do(proxyReq)
```

Comment:
This could close `retry.Body` or start mid-stream on the next attempt. If we want a retry policy, we'll need to create a copy of the request body.

Reply:
copied the body to `requestBody`